server/app.js

```diff
@@ -1,5 +1,50 @@
+const fs = require('fs').promises;
 const express = require('express');
 const app = express();
+app.use(express.json());
+app.use(express.static('../client/build'));
+const path = './data.json';
+app.get('/api/tickets', async (req, res) => {
+  const tickets = await fs.readFile(path);
+  const ticketsJson = JSON.parse(tickets);
+  if (!req.query.searchText) {
+    res.send(ticketsJson);
+  } else {
+    const ticketsArray = ticketsJson.filter((item) => item.title.toLowerCase().includes(req.query.searchText.toLowerCase()));
+    res.send(ticketsArray);
+  }
+});
 
+app.post('/api/tickets/:ticketId/done', async (req, res) => {
+  const tickets = await fs.readFile(path);
+  let ticketsJson = JSON.parse(tickets);
+  try {
+    const ticketsArray = ticketsJson.map((item) => {
```

Comment:
What happens if an invalid `ticketId` is requested? How are you handling that?
👨🏻‍🏫 - prepare 5 min explanation

Reply:
Fixed and prepared thank you.